src/components/structures/MatrixChat.js

```diff
@@ -364,7 +364,7 @@ module.exports = React.createClass({
                 // is completed in another browser, we'll be 401ed for using
                 // a guest access token for a non-guest account.
                 // It will be restarted in onReturnToGuestClick
-                Lifecycle.stopMatrixClient();
+                Lifecycle.stopMatrixClient(false);
```

Comment:
we were only clearing half the stores, to be honest, so passing `true` here would have been a bigger change in behaviour.

As I understand it, Dave added this line of code solely to stop the MatrixClient polling sync, to avoid the 401. That's what the comment says. It doesn't say anything about clearing the stores. Indeed, we can't clear the stores here, in case you want to go back to being a guest.

I think it was pretty broken before, and as I understand it, @lukebarnard1 is making guest access go away, so I'm reluctant to block the work I'm trying to do by spending a lot of time fixing the upgrading-from-guest flow.